feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/actioncreator/SessionContentsActionCreator.kt

```diff
@@ -66,27 +67,39 @@ class SessionContentsActionCreator @Inject constructor(
     }
 
     fun toggleFavorite(session: Session) {
-        launch {
-            try {
+        fx {
+            !effect {
```

Comment:
What's the purpose to use fx? 🤔 
AFAIK, fx forces to represent side-effects by using suspend functions through compilation errors.

If you want to just integrate with coroutine, then arrow supports coroutine extension with DefferedK

```
val defferedK = async {
...
LoadingState.LOADED
}.k().handleErrorWith {
   dispatcher.dispatch(
                        Action.ShowProcessingMessage(
                            Message.of(
                                R.string.session_favorite_connection_error
                            )
                        )
                    )
  LoadingState.INITIALIZED
 }

dispatcher.dispatch(defferedK.unsafeAttemptSync())
```

Reply:
😇 
https://github.com/arrow-kt/arrow/blob/master/modules/effects/arrow-effects-kotlinx-coroutines-extensions/src/main/kotlin/arrow/effects/coroutines/extensions/deferredk.kt

Reply:
Yes. Probably .k or DefferedK will be removed. 😇

Reply:
After diving into arrow fx, I found a cause.

The current *fx* is for Try monad and it doesn't take care of asynchronous operations. 

In FP world, IO monad is what you need to use.

- Add *io.arrow-kt:arrow-effects-io-extensions* dependency
- Import *arrow.effects.extensions.io.fx.fx* correctly

Then, follow the style below.

```
fun toggleFavorite(...) {
        unsafe {
            runNonBlocking({
                fx {
                    ....
                }
            }, { // Either monad will be coming
            })
        }
}
```